Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -0,0 +1,687 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import (
+    DataProcessorAlgorithm,
+    AlgorithmFactory,
+    MatrixWorkspaceProperty,
+    IPeaksWorkspaceProperty,
+    WorkspaceUnitValidator,
+    FileProperty,
+    FileAction,
+    Progress,
+)
+from mantid.kernel import (
+    Direction,
+    FloatBoundedValidator,
+    IntBoundedValidator,
+    StringListValidator,
+    EnabledWhenProperty,
+    PropertyCriterion,
+)
+import numpy as np
+from scipy.ndimage import uniform_filter
+from scipy.signal import convolve
+from IntegratePeaksSkew import InstrumentArrayConverter, get_fwhm_from_back_to_back_params
+from FindSXPeaksConvolve import make_kernel, get_kernel_shape
+from enum import Enum
+
+
+class PEAK_STATUS(Enum):
+    WEAK = "Weak peak"
+    STRONG = "Strong peak"
+    ON_EDGE = "Peak mask is on the detector edge"
+    ON_EDGE_INTEGRATED = "Peak mask is on the detector edge"
+    NO_PEAK = "No peak detected."
+
+
+class WeakPeak:
+    def __init__(self, ipk, ispec, tof, tof_fwhm, ipks_near):
+        self.ipk = ipk
+        self.ispec = ispec  # spectrum at center of kernel that corresponds to max I/sigma from convolution
+        self.tof = tof  # TOF at center of kernel that corresponds to max I/sigma from convolution
+        self.tof_fwhm = tof_fwhm
+        self.ipks_near = ipks_near
+
```

Comment:
Can I suggest refactoring this function with a `@dataclass` decorator? Since it's only a class for containg different data types, using the decorator may improve readability and is not necessary to add a constructor (it's more like a matlab struct or a named tuple)
```suggestion
@dataclass
class WeakPeak:
        ipk: int
        ispec: int  # spectrum at center of kernel that corresponds to max I/sigma from convolution
        tof: float  # TOF at center of kernel that corresponds to max I/sigma from convolution
        tof_fwhm: float = 0.0
        ipks_near: list? #not sure what is the type

```

Reply:
Not come accros sthis `dataclass` thing thanks for showing me! It is indeed a lot more readable